### ISSUE TITLE
CORS headers, added by Steroids, only work via $ steroids connect

### ISSUE THREAD

**Issue by Leeds-eBooks** [AppGyver/steroids#902]
Forum thread [here](https://forums.appgyver.com/#!/steroids:ajax-request-works-in-emula).

Getting this error on deployed builds, not on `$ steroids connect`:
```
XMLHttpRequest cannot load http://mydomain.com/rest/login. Origin http://localhost is not allowed by Access-Control-Allow-Origin.
```

If I had to guess, I would say that the trickery that AppGyver use to add CORS headers to all incoming requests is only working on builds generated by `$ steroids connect`.

The Origin request header from both `deploy`ed and `connect` builds is `http://localhost`, so the request sent out from the app is the same, and therefore the response from the server must be the same. The difference is happening at the point at which the app receives the response from the server – on `connect` builds, the required CORS headers are added, on `deployed` builds, they are not.

I can make a minimum reproduction if necessary, but as I was able to debug the requests and get a clear error message, it's probably not necessary? The only difference between a working case and a failing case is whether the app is being served via `connect` or if it has been deployed. Perhaps the `network.extraResponseHeaders` settings in `app.coffee` are not getting uploaded correctly on `deploy`?

**Comment by Leeds-eBooks** [comment]
Fortunately I have access to the server, so I've just added CORS headers to the responses from the server and everything seems to be working fine. I'm sure this will still be an issue for many people if what works on one type of build doesn't work on another.

**Comment by harsaharsa** [comment]
Thanks for reporting this – we'll investigate; one other user reported a similar issue also.

**Comment by martinbroeckl** [comment]
I have the same issue. Are there any news so far?

**Comment by Leeds-eBooks** [comment]
I think this has been fixed, but not pushed to an update yet.

Sent using CloudMagic [https://cloudmagic.com/k/d/mailapp?ct=ti&cv=6.0.64&pv=8.2]
On Tue, Jun 30, 2015 at 07:04, martinbroeckl <notifications@github.com> wrote:
I have the same issue. Are there any news so far?

—
Reply to this email directly or view it on GitHub
[https://github.com/AppGyver/steroids/issues/902#issuecomment-116991295] .[https://github.com/notifications/beacon/AELX77vKLrESfwFg-lSNHvetCckcB56aks5oYij_gaJpZM4D-3j7.gif]

**Comment by harsaharsa** [comment]
This should'vbe been fixed with Steroids CLI v4.1.9. Can you report if the problem still persists?

**Comment by nileshdarade** [comment]
I'm on 
steroids --version
AppGyver Steroids² 4.1.23

In Android I'm getting this error:

XMLHttpRequest cannot load https://host/accesstoken?grant_type=client_credentials. A wildcard '*' cannot be used in the 'Access-Control-Allow-Origin' header when the credentials flag is true. Origin 'http://localhost' is therefore not allowed access.

On IOS is working perfectly fine.
